pmd-java/etc/grammar/Java.jjt

```diff
@@ -1801,12 +1804,13 @@ void PostfixExpression() #PostfixExpression((jjtn000.getImage() != null)):
   PrimaryExpression() [ "++" {jjtThis.setImage("++");} | "--" {jjtThis.setImage("--");} ]
 }
 
-void CastExpression() #CastExpression(>1):
+void CastExpression() :
 {}
 {
-  LOOKAHEAD("(" (Annotation())* Type() ")") "(" (Annotation() {checkForBadTypeAnnotations();})* Type() ")" UnaryExpression()
-| LOOKAHEAD("(" (Annotation())* Type() "&") "(" (Annotation() {checkForBadTypeAnnotations();})* Type() ( "&" {checkForBadIntersectionTypesInCasts(); jjtThis.setIntersectionTypes(true);} ReferenceType() )+ ")" UnaryExpressionNotPlusMinus()
-| "(" (Annotation() {checkForBadTypeAnnotations();})* Type() ")" UnaryExpressionNotPlusMinus()
+  LOOKAHEAD(
+    "(" (Annotation())* PrimitiveType() ")"
+  ) "(" (Annotation() {checkForBadTypeAnnotations();})* PrimitiveType() ")" UnaryExpression()
```

Comment:
If I use `Type()` here instead of `PrimitiveType()` the generated parser lookahead code will check for `Type()` regardless of what the `LOOKAHEAD` says on the previous line... not sure why, possibly a bug on JavaCC?